plugins/webp-uploads/helper.php

```diff
@@ -348,3 +354,32 @@ function webp_uploads_should_discard_additional_image_file( array $original, arr
 	}
 	return false;
 }
+
+/**
+ * Checks if a mime type is supported by the server.
+ *
+ * Includes special handling for false positives on AVIF support.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $mime_type The mime type to check.
+ * @return bool Whether the server supports a given mime type.
+ */
+function webp_uploads_mime_type_supported( string $mime_type ): bool {
+	if ( ! wp_image_editor_supports( array( 'mime_type' => $mime_type ) ) ) {
+		return false;
+	}
+
+	// In certain server environments Image editors can report a false positive for AVIF support.
+	if ( 'image/avif' === $mime_type ) {
+		$editor = _wp_image_editor_choose( 'image/avif' );
+		if ( 'WP_Image_Editor_GD' === $editor ) {
+			return function_exists( 'imageavif' );
+		}
+		if ( 'WP_Image_Editor_Imagick' === $editor ) {
```

Comment:
```suggestion
		if ( $editor instanceof WP_Image_Editor_Imagick ) {
```

Reply:
`_wp_image_editor_choose` returns a string, not the editor class

Reply:
My bad. In that case, per above:

```suggestion
		if ( is_a( $editor, WP_Image_Editor_Imagick::class, true ) && class_exists( 'Imagick' ) ) {
```